### PR TITLE
[WIP] Partially replace Bluebird with native Promises.

### PR DESCRIPTION
**Pull request by kibertoad**
Remove deprecated global methods. Remove Bluebird from typings.

Does not remove Bluebird as a dependency completely, only replaces Bluebird with native Promises where it was straightforward to do so without additional code changes. Idea is to grab low-hanging fruits first and then split remaining (non-trivial) work in chunks and do it incrementally (there were some devs in the community who offered their help as well).

### CODE REVIEW

**Comment by kibertoad** [comment]
@lorefnon Removal of bluebird from typings causes major problems with our typings. Our genericified types seem to rely heavily on usage of `.map()` and `.reduce()` methods. How would you recommend to proceed? Simply drop part of the tests that cover that part and rely on users doing mapping on their side from `any` returned by knex?

File: package.json [hunk 1]
```diff
@@ -79,7 +78,8 @@
     "coveralls": "nyc report --reporter=text-lcov | coveralls",
     "lint": "eslint \"src/**/*.js\" \"test/**/*.js\"",
     "lint:types": "dtslint types",
-    "pretest": "npm run lint && npm run lint:types",
+    "old:pretest": "npm run lint && npm run lint:types",
```
Comment (kibertoad) [inline package.json]: this is temporary, I just don't want CI to fail due to an already known problem with typings, will be restored after we figure out how to proceed.

**Comment by lorefnon** [comment]
> Our genericified types seem to rely heavily on usage of .map() and .reduce() methods. How would you recommend to proceed? Simply drop part of the tests that cover that part and rely on users doing mapping on their side from any returned by knex?

I just moved the map/reduce etc. outside of await in https://github.com/tgriesser/knex/pull/3251 which should fix this.

File: test/tape/pool.js [hunk 2]
```diff
@@ -1,12 +1,12 @@
 'use strict';
 
 const test = require('tape');
+const Promise = require('bluebird');
```
Comment (elhigu) [inline test/tape/pool.js]: Is this right?

**Comment by elhigu** [comment]
So far did seem like just removing bluebird requires and fixing types. I hope fixing tests and docs will be as straightforward.

**Comment by elhigu** [comment]
maybe we could stop passing Promise to migrations and seeds.

File: test/tape/pool.js (hunk 2 above)
Comment (kibertoad) [reply test/tape/pool.js]: yeah, removed it at first and then realized it makes tests fail. should revert change altogether

File: package.json [hunk 3]
```diff
@@ -19,6 +18,7 @@
     "liftoff": "3.1.0",
     "lodash": "^4.17.11",
     "mkdirp": "^0.5.1",
+    "oracledb": "^3.1.2",
```
Comment (caiquecastro) [inline package.json]: Is it correct?

Comment (kibertoad) [reply package.json]: nope. good catch! will remove.

**Comment by kibertoad** [comment]
Useful parts of this were merged separately, closing this one.

**Pull request closed without merging**